plugins/main/public/react-services/navigation-service.tsx

```diff
@@ -124,6 +124,34 @@ class NavigationService {
     return getCore().application.getUrlForApp(appId, options);
   }
 
+  public switchTab(newTab: any, force: any) {
+    const urlSearchParams = new URLSearchParams(this.getSearch());
+    urlSearchParams.set('tab', newTab);
+    this.navigate(
+      `${this.getPathname()}?${urlSearchParams
+        .entries()
+        .reduce(
+          (total, current) =>
+            (total += `${total ? '&' : ''}${current[0]}=${current[1]}`),
+          '',
+        )}`,
+    );
+  }
+
+  public switchSubTab = (subTab: string) => {
+    const urlSearchParams = new URLSearchParams(this.getSearch());
+    urlSearchParams.set('tabView', subTab);
+    this.navigate(
+      `${this.getPathname()}?${urlSearchParams
+        .entries()
+        .reduce(
+          (total, current) =>
+            (total += `${total ? '&' : ''}${current[0]}=${current[1]}`),
+          '',
+        )}`,
+    );
```

Comment:
It's the same case as the previous comment.

Reply:
It was decided to make a solution that fits all cases and in one of the methods to use the 'Array from' suggested